course3/homework-vector/vector.cc

```diff
@@ -1,9 +1,55 @@
 #include <iostream>
 #include <cassert>
+#include <cmath>
 
 class Vector
 {
-    // TODO
+    private:
+        int private_x, private_y;
+
+    public:
+        // required constructors
+        Vector(int x, int y): private_x(x), private_y(y) {}
+        Vector() {
+            private_x = 0;
+            private_y = 0;
```

Comment:
Look what's happening here: the member variables are first _default initialized_, and then you _assign_ to them.
I case of integer the first one is a no-op, but just for the sake of excercise: can you do it w/o assignment?

Also: can you make the object default-constructible with known default values of the fields without implementing the default constructor at all?